web/src/list_widget.ts

```diff
@@ -17,6 +17,31 @@ type ListWidgetMeta<Key = unknown, Item = Key> = {
     $scroll_container: JQuery;
 };
 
+type ListWidgetOpts<Key = unknown, Item = Key> = {
+    name?: string;
+    get_item: (key: Key) => Item;
+    modifier: (item: Item) => string;
+    init_sort?: string | SortingFunction<Item>;
+    initially_descending_sort?: boolean;
+    html_selector?: (item: Item) => JQuery;
```

Comment:
In addition to `(item: Item) => Jquery`, `(item: Item) => string` is also acceptable.

Reply:
I took a look and it seems there was only one place where `html_selector` was returning string. So I think it makes sense to return a `JQuery` collection there instead of string and avoid writing type like `string | JQuery` ?
This is the one place where we are returning string(settings_users.js) - 
https://github.com/zulip/zulip/blob/784622ee5d3e5bca4db9ebafba748acd797b427d/web/src/settings_users.js#L290

Reply:
Yeah, also it seems that `JQuery` instead of `string` is expected from `html_selector`:
https://github.com/zulip/zulip/blob/d6c21c2237918f402988bd7e9463afa9253a1192/web/src/list_widget.js#L395-L401
I'm not sure if this works previously for `settings_users` and whether it was an intentional decision to use a `string` there.

Reply:
Yeah I think that `insert_rendered_row` was written only for use in `recent_topics_ui` and I don't think that this method previously worked for `settings_users`.